registries/adapters/dockerhub_adapter.go

```
@@ -265,12 +277,15 @@ func (r DockerHubAdapter) getBearerToken(imageName string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	defer response.Body.Close()
+	defer func() {
```

Are we doing this because we added the error check to the `check` command in the makefile?

I actually don't like adding these go routines just because of a tool. Wondering we can just add this to the make file, and not add it to CI. The tool is useful for checking the code, but I don't know if we want to add this code when we really do want to ignore the error.

---

When you say adding go routines, I am not sure I follow do you mean the additional function call as we are really are just deferring one function call over another. Originally we were deferring the Close method where as now we are deferring an anonymous function that calls the ```Close``` method. There are no additional  go routines created?

I know where you are coming from with ignoring the error when closing the response body, but do we actually want to ignore it? What is the downside to not ignoring it?
I do think there is value in having the err check tool run as part of the PR process as it gives us the confidence that any new code has at least handled any possible errors. I think a couple of additional err checks as a cost is not that high

---

the defer func() solely for errcheck is annoying because you really don't need to check the err on a close() it just gets ignored. What are we going to do if the response body doesn't get closed? If this ends up logging a bunch of warnings to the broker log because of errors that we would've ignored before I will ask we remove these. For now, I'll let it stand.